### DISCUSSION TITLE
Visualizing Packet Drops - Showcase

### DISCUSSION THREAD

**Discussion by rhornig** [inet-framework/inet#583] [Showcases]
Discuss the "Packet Drops" visualization showcases here. Ideas and recommendations are welcome.

**Comment by ManiAm** [comment]
The 'MAC retry limit reached' scenario throws run-time error. 

'NF_PACKET_DROP' is emitted from the originatorProcessFailedFrame method in the Dcf.cc file. The receiveSignal of the PacketDropVisualizerBase.cc received the signal, but does not recognize it! 

The previous scenario works fine. The arpResolutionTimeOut method in the IPv4.cc file emits signal LayeredProtocolBase::packetFromUpperDroppedSignal and can be decoded successfully.

Apparently there is inconsistency between the drop packet signals in different modules or it has not been added into the 'or' expression of receiveSignal.

**Comment by levy** [comment]
Thanks for the bug report!

The packet drop signalling and visualization mechanism has been refactored in the upcoming INET 4.0 version. There's only one packetDropped signal now emitting the packet and an additional details object with drop reason and other parameters. Therefore the condition in the visualizer became trivial. I'm not sure if we want to fix this in the 3.x branch.
